pkg/analysis/analyzer.go

```diff
@@ -159,7 +159,7 @@ func (a *Analyzer) criteriaRoutines(
 	}
 }
 
-func (a *Analyzer) Start(notifications <-chan entities.NodesGatheringNotification) <-chan entities.Alert {
+func (a *Analyzer) Start(notifications <-chan entities.NodesGatheringNotification) chan entities.Alert {
```

Comment:
I don't see a problem here; another function will also be writing into `Alerts`, why do I need to create several channels?

Reply:
Yep, writing into the channel in one function is better. Conventionally, the _writer channel_ should be closed when write has finished his routine task. When you write to one channel from multiple goroutines you can close the channel at least two times. This will lead to a `panic` state.